network/newstream/wire.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package newstream
```

Comment:
The package is called `newstream`. Probably we should not be prefixing most of the structs in this package with `Stream`.

Reply:
i'm not sure i understand. existing stream directory will be removed. the package is not the same package. do you want the types to be prefixed with `NewStream`?

Reply:
when you package is called `stream` or `newstream`, the types in side already have the context of the word / concept `Stream`, so no need to generally prefix them. For example `newstream.StreamProvider` -> `newstream.Provider`.

Reply:
fair enough :+1: I'll take care of it in the following iteration